lib/langchain/base_conversation_memory.rb

```diff
@@ -27,10 +24,20 @@ def add_examples(examples)
 
     def append_ai_message(message)
       @messages << {role: "ai", content: message}
+      after_add_message(message, "ai")
     end
 
     def append_user_message(message)
       @messages << {role: "user", content: message}
+      after_add_message(message, "user")
+    end
+
+    def after_add_message(message, role)
```

Comment:
let's not create empty methods in langchain's class. It's better to do:
```ruby
def append_ai_message(content)
  append_message {role: "ai", content: content}
end

def append_message(message)
  @messages << message
end
```

and then in subclass you can override append_message:
```ruby
def append_message(message)
  super

  # do stuff here 
end
```